Feat/supabase

OsLe99 opened the pull request:
Bytt från mongodb till supabase

OsLe99 committed:
Updated database service to Supabase. Currently works for only certain tasks
Create booking
Update booking

OsLe99 committed:
Info and contacts can now be shown on InfoPage

OsLe99 committed:
Fixed error in unit test

OsLe99 committed:
Solved merge conflict

OsLe99 committed:
Fixed picked dates not displayed in the calendar on booking step 3

ConsidMadrusan commented on SnapphaneScoutDistriktBookingApp/MauiProgram.cs:
```diff
@@ -21,6 +22,18 @@ public static MauiApp CreateMauiApp()
                 });
+            // Supabase
+            var url = "https://cwudkbwltvsjesoksxno.supabase.co";
+            var key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImN3dWRrYndsdHZzamVzb2tzeG5vIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjAwMTA1MjIsImV4cCI6MjA3NTU4NjUyMn0.1uUS471huzB5OtJaJsQCxkGItcLMMekzwMz0m3rH1LY";
```
Sätt upp detta som miljövariablar

OsLe99 committed:
Removed secrets
Replaced with env variables

ConsidMadrusan commented on SnapphaneScoutDistriktBookingApp/Services/BookingService.cs:
```diff
@@ -23,28 +23,28 @@ public BookingService(IDbService db, IEmailService emailService)
-            await _emailService.SendEmailAsync("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", "emil.berg@campusnykoping.se", customer.Email, customer);
-            var newBooking = await FindAddedBookingByIdAsync(customer);
+            await _db.AddCustomerAsync(booking);
+            await _emailService.SendEmailAsync("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", "emil.berg@campusnykoping.se", booking.Email, booking);
```
Detta kan också ändras till miljövariablar för att t.ex. kunna byta från vilken e-post mailen ska komme ifrån

ConsidMadrusan commented on SnapphaneScoutDistriktBookingApp/Services/EmailService.cs:
```diff
@@ -13,36 +13,36 @@ namespace SnapphaneScoutDistriktBookingApp.Services
     public class EmailService : IEmailService
     {
-        public async Task SendEmailAsync(string apiKey, string fromEmail, string toEmail, Models.Customer customer)
+        public async Task SendEmailAsync(string apiKey, string fromEmail, string toEmail, Models.Booking booking)
```
Försök att fixa importen så att man man bara har Booking och inte Models.Booking

ConsidMadrusan commented on SnapphaneScoutDistriktBookingApp/Services/EmailService.cs:
```diff
@@ -13,36 +13,36 @@ namespace SnapphaneScoutDistriktBookingApp.Services
-        public async Task SendEmailAsync(string apiKey, string fromEmail, string toEmail, Models.Customer customer)
+        public async Task SendEmailAsync(string apiKey, string fromEmail, string toEmail, Models.Booking booking)
         {
             toEmail = "oscar.lejon@campusnykoping.se";
```
Sätt upp en variabel för att toggla testmail

ConsidMadrusan commented on SnapphaneScoutDistriktBookingApp/ViewModels/InfoPageViewModel.cs:
```diff
@@ -68,14 +98,16 @@ private async Task UpdateInfoDBAsync()
         {
             Models.Info info = new Models.Info()
```
Fixa importen till Info enbart

OsLe99 committed:
Removed exposed secrets
Sendgrid api key and email removed from code and replaced with environment variables.

OsLe99 committed:
Minor code fixes.
Removed Models prefix.

ConsidMadrusan commented on SnapphaneScoutDistriktBookingApp/Views/ViewBooking.xaml.cs:
```diff
@@ -49,12 +49,12 @@ private async void OnSearchClicked(object sender, EventArgs e)
             if (booking != null)
             {
-                ResultsCollectionView.ItemsSource = new List<Customer> { booking };
+                ResultsCollectionView.ItemsSource = new List<Models.Booking> { booking };
```
Import bör fixas

OsLe99 committed:
Added appsettings.json for keys instead of enviroment variables

OsLe99 committed:
Fixed import of Booking.
Uses BookingModel instead.

OsLe99 committed:
Removed tracking of local appsettings file

OsLe99 committed:
Fixed .gitignore
